Model/Quote/Address/Total/Surcharge.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * @by SwiftOtter, Inc. 3/30/17
+ * @website https://swiftotter.com
+ **/
+
+namespace SwiftOtter\ShippingSurcharge\Model\Quote\Address\Total;
+
+use \SwiftOtter\ShippingSurcharge\Model\Surcharge as SurchargeModel;
+
+class Surcharge extends \Magento\Quote\Model\Quote\Address\Total\AbstractTotal
+{
+    /**
+     * @var \Magento\Framework\Pricing\PriceCurrencyInterface
+     */
+    private $priceCurrency;
+
+    /**
+     * @var \SwiftOtter\ShippingSurcharge\Api\Calculator\SurchargeCalculatorInterface
+     */
+    private $surchargeCalculator;
+
+    /**
+     * @var \Magento\Quote\Api\Data\TotalSegmentExtensionFactory
+     */
+    private $cartExtensionFactory;
+
+
+    public function __construct(
+        \Magento\Framework\Pricing\PriceCurrencyInterface $priceCurrency,
+        \SwiftOtter\ShippingSurcharge\Api\Calculator\SurchargeCalculatorInterface $surchargeCalculator,
+        \Magento\Quote\Api\Data\CartExtensionFactory $cartExtensionFactory
+    )
+    {
+        $this->setCode(SurchargeModel::SURCHARGE);
```

Comment:
Here's a place where it might be good to add another line, to delineate the method call from the property assignments.